versa/utterance_metrics/noresqa.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+# Copyright 2024 Jiatong Shi
+#  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
+
+
+import os
+import numpy as np
+import librosa
+import torch
+import sys
+
+import logging
+from urllib.request import urlretrieve
+import torch.nn as nn
+
+sys.path.append("../../")
+sys.path.append("./checkpoints/Noresqa")
+
+try:
+    import fairseq
+except ImportError:
+    logging.warning(
+        "fairseq is not installed. Please use `tools/install_fairseq.sh` to install"
+    )
+
+try:
+    from tools.Noresqa.utils import (
+        feats_loading,
+        model_prediction_noresqa,
+        model_prediction_noresqa_mos,
+    )
+    from tools.Noresqa.model import NORESQA
+
+except ImportError:
+    logging.warning(
+        "noresqa is not installed. Please use `tools/install_noresqa.sh` to install"
+    )
+    Noresqa = None
+
+sys.path.append("./checkpoints/Noresqa")
+
+
+def noresqa_model_setup(model_tag="default", metric_type=0, use_gpu=False):
+    if use_gpu:
+        device = "cuda"
+    else:
+        device = "cpu"
+
+    if model_tag == "default":
+
+        if not os.path.isdir("./checkpoints/Noresqa"):
```

Comment:
Is it possible to set the cache dir?

I may misunderstood, but if it is wav2vec_small, there are some similar models using it. We may redesign the cache for those in a later timing.

Reply:
We can set the cache dir by using `urlretrieve`. Yeah we can redesign the cache later~